File: implementations/node/sdk/src/contracts/observation/execution-step-status.ts
export const AI_SDK_EXECUTION_STEP_STATUSES = [
  "Pending",
  "Ready",
  "Running",
  "WaitingForRetry",
  "WaitingForExternal",
  "Completed",
  "Failed",
] as const;

export type AiSdkExecutionStepStatus = (typeof AI_SDK_EXECUTION_STEP_STATUSES)[number];

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/observation/execution_step_status.py
from enum import Enum


class AiSdkExecutionStepStatus(str, Enum):
    PENDING = "Pending"
    READY = "Ready"
    RUNNING = "Running"
    WAITING_FOR_RETRY = "WaitingForRetry"
    WAITING_FOR_EXTERNAL = "WaitingForExternal"
    COMPLETED = "Completed"
    FAILED = "Failed"
